components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -2722,6 +2755,52 @@ public static AuthenticatedUser createAuthenticatedUser(String username, String
         return authenticatedUser;
     }
 
+    /**
+     * Creates an instance on AuthenticatedUser{@link AuthenticatedUser} for the given parameters.
+     *
+     * @param username        username of the user
+     * @param userStoreDomain user store domain
+     * @param tenantDomain    tenent domain
+     * @param idpName    idp name
+     * @return an instance of AuthenticatedUser{@link AuthenticatedUser}
+     */
+    public static AuthenticatedUser createAuthenticatedUser(String username, String userStoreDomain, String
+            tenantDomain, String idpName) {
+
+        AuthenticatedUser authenticatedUser = new AuthenticatedUser();
+        authenticatedUser.setUserName(username);
+        authenticatedUser.setTenantDomain(tenantDomain);
+
+        if (OAuth2ServiceComponentHolder.isIDPIdColumnEnabled()) {
+            if (!StringUtils.equals(FrameworkConstants.LOCAL_IDP_NAME, idpName) && !OAuthServerConfiguration.getInstance()
+                    .isMapFederatedUsersToLocal()) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Found federated idp: " + idpName + " for user: " + username + " in " +
+                            "tenant domain:" + tenantDomain + ". Flag user as a federated user.");
+                }
+                authenticatedUser.setFederatedUser(true);
+                authenticatedUser.setFederatedIdPName(idpName);
+            } else {
+                authenticatedUser.setUserStoreDomain(userStoreDomain);
+            }
+        } else {
+            // Preserving behaviour in deprecated createAuthenticatedUser method which did not have the idpName.
+            if (StringUtils.startsWith(userStoreDomain, OAuthConstants.UserType.FEDERATED_USER_DOMAIN_PREFIX) &&
+                    !OAuthServerConfiguration.getInstance().isMapFederatedUsersToLocal()) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Federated prefix found in domain: " + userStoreDomain + " for user: " + username + " in " +
```

Comment:
Let's keep the charactor count to 120 per line.

Reply:
Fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1128